### PR TITLE
Compact dashboard and make layout configurable

### PR DESCRIPTION
**Pull request by eoinsha**

### CODE REVIEW

File: serverless-plugin/dashboard.js [hunk 1]
```diff
@@ -60,7 +58,7 @@ module.exports = function dashboard (serverless, config, context) {
       ...stateMachineWidgets,
       ...lambdaWidgets
     ])
-    const dash = { start: DASHBOARD_PERIOD, widgets: positionedWidgets }
+    const dash = { start: config.timeRange, end: config.timeRange.end, widgets: positionedWidgets }
```
Comment (lmammino) [inline serverless-plugin/dashboard.js:61]: Shouldn't this be as follows? (see `.start`)

```suggestion
    const dash = { start: config.timeRange.start, end: config.timeRange.end, widgets: positionedWidgets }
```

File: README.md [hunk 2]
```diff
@@ -34,6 +34,13 @@ Supported options along with their defaults are shown below.
 custom:
   slicWatch:
     topic: SNS_TOPIC_ARN
+    dashboard:
+      timeRange:
+        start: '-PT3H'
```
Comment (lmammino) [inline README.md:39]: do we need also an `end`?

File: serverless-plugin/default-config.js [hunk 3]
```diff
@@ -1,7 +1,16 @@
 'use strict'
 
 module.exports = {
-  dashboard: {},
+  dashboard: {
+    timeRange: {
+      start: '-PT3H'
```
Comment (lmammino) [inline serverless-plugin/default-config.js:6]: here too: do we need `end`?

File: serverless-plugin/tests/dashboard.test.js [hunk 4]
```diff
@@ -6,7 +6,17 @@ const CloudFormationTemplate = require('../cf-template')
 const { test } = require('tap')
 
 const sls = {}
-const config = {}
+const config = {
+  timeRange: {
+    start: '-PT3H'
```
Comment (lmammino) [inline serverless-plugin/tests/dashboard.test.js:11]: same here

**Review by lmammino — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (eoinsha) [reply README.md:39]: Not for the defaults, since end is optional and usually unused.

**Pull request merged**